### PR TITLE
Display name population

### PR DESCRIPTION
**Pull request by stevomcallister**

### CODE REVIEW

File: src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/config/TestWebClientConfiguration.kt
Comment (stevomcallister) [inline src/integrationTest/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/config/TestWebClientConfiguration.kt:1]: Don't need this hack any more as I've implemented auth in the tests.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/client/manageusers/ManageUsersApiClient.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.client.manageusers
+
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import reactor.core.publisher.Mono
+
+@Component
+class ManageUsersApiClient(@Qualifier("manageUsersApiWebClient") private val manageUsersApiClient: WebClient) {
```
Comment (stevomcallister) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/client/manageusers/ManageUsersApiClient.kt:10]: Client to get the user details - display name for a username. Note that the endpoint only needs a valid token - no additional roles are needed. So no HAAR updates.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/mapper/conversation/ConversationsResourceMapper.kt
Comment (stevomcallister) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/mapper/conversation/ConversationsResourceMapper.kt:1]: Changed mapper from mapstruct to manual mapping.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/ConversationController.kt [hunk 2]
```diff
@@ -29,6 +29,7 @@ import java.util.UUID
 class ConversationController(
   private val conversationService: ConversationService,
   private val conversationMapper: ConversationsResourceMapper,
+  private val newConversationMapper: ConversationsResourceMapper,
```
Comment (nathanrussell-moj-digital) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/ConversationController.kt:32]: Not sure I follow this ... does `ConversationResourceMapper` need injecting twice? 🤔  Its the same mapper class isnt it?
In fact, does this controller need changing at all? 🤔

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/mapper/conversation/ConversationsResourceMapper.kt [hunk 3]
```diff
@@ -1,45 +1,76 @@
...
-  ],
-  imports = [
-    Instant::class,
-  ],
-)
-interface ConversationsResourceMapper {
-  @Mapping(source = "request.prisonId", target = "note.prisonId")
-  @Mapping(source = "request.note", target = "note.content")
-  fun toCreateConversationDto(request: CreateConversationRequest, prisonNumber: String): CreateConversationDto
+@Component
```
Comment (nathanrussell-moj-digital) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/mapper/conversation/ConversationsResourceMapper.kt:19]: 👍 Another mapstruct mapper bites the dust 🤣 👍

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/ConversationController.kt (hunk 2 above)
Comment (stevomcallister) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/ConversationController.kt:32]: No that a mistake - I wrote the newMapper while I was writing it and then deleted the old one. Intellij has refactored and I've forgotten to remove the old one

**Review by nathanrussell-moj-digital — APPROVED** [review]
LGTM 👍

**Pull request merged**